engine/proofview.ml

```diff
@@ -41,7 +41,7 @@ let compact el ({ solution } as pv) =
   let nf c = Evarutil.nf_evar solution c in
   let nf0 c = EConstr.(to_constr ~abort_on_undefined_evars:false solution (of_constr c)) in
   let size = Evd.fold (fun _ _ i -> i+1) solution 0 in
-  let new_el = List.map (fun (t,ty) -> nf t, nf ty) el in
+  let new_el = { entry = List.map (fun (evk,t,ty) -> evk, nf t, nf ty) el.entry } in
   let pruned_solution = Evd.drop_all_defined solution in
```

Comment:
Since we drop the defined evars we end up referring to undeclared evars

Reply:
Hmpf, this is annoying. I'll have to also adapt the evar dropping API...